core/src/test/scala/com/softwaremill/events/EventMachineTest.scala

```diff
@@ -90,6 +94,25 @@ class EventMachineTest extends FlatSpec with Matchers with SqlSpec with Eventual
       m.asyncEventRunner.stop()
     }
   }
+
+  it should "recover stored events" in {
+    // given
+    var actions = Vector.empty[String]
+
+    val m = createModules(Registry()
+      .registerModelUpdate[Event1] { e => DBIOAction.successful(()).map { r => actions :+= "mu1" + e.data.data; r } }
+      .registerEventListener[Event1] { e => DBIOAction.successful(Nil).map { r => actions :+= "el1"; r } })
+
+    implicit val hc = HandleContext.System
+    val result = m.eventMachine.run(CommandResult.successful((), Event(Event1("x")).forNewAggregate)).futureValue
+
+    // when
+    val failOverFuture = m.eventMachine.failOverFromStoredEventsTo().futureValue
```

Comment:
Sorry, I meant that it's possible to have an event, say, BCreate, which creates an entity of  type B derived from an entity of type A. There could be other, later events on the entity of type A. There's a clear ordering, but it could be useful for debugging to re-create the view model of the A entity as of the B entity creation.